src/ape/managers/project.py

```diff
@@ -452,85 +553,116 @@ def __getattr__(self, attr_name: str) -> ContractContainer:
             _converter=self.converter,
         )
 
-    @property
-    def interfaces_folder(self) -> Path:
+    def extensions_with_missing_compilers(self, extensions: Optional[List[str]]) -> List[str]:
         """
-        The path to the ``interfaces/`` directory of the project.
+        All file extensions in the ``contracts/`` directory (recursively)
+        that do not correspond to a registered compiler.
+
+        Args:
+            extensions (List[str], optional): If provided, returns only extensions that
+                are in this list. Useful for checking against a subset of source files.
 
         Returns:
-            pathlib.Path
+            List[str]: A list of file extensions found in the ``contracts/`` directory
+            that do not have associated compilers installed.
         """
+        extensions_found = []
 
-        return self.path / "interfaces"
+        def _append_extensions_in_dir(directory: Path):
+            for file in directory.iterdir():
+                if file.is_dir():
+                    _append_extensions_in_dir(file)
+                elif (
+                    file.suffix
+                    and file.suffix not in extensions_found
+                    and file.suffix not in self.compilers.registered_compilers
+                ):
+                    extensions_found.append(file.suffix)
 
-    @property
-    def scripts_folder(self) -> Path:
-        """
-        The path to the ``scripts/`` directory of the project.
+        _append_extensions_in_dir(self.contracts_folder)
+        if extensions:
+            extensions_found = [e for e in extensions_found if e in extensions]
 
-        Returns:
-            pathlib.Path
+        return extensions_found
+
+    def lookup_path(self, key_contract_path: Path) -> Optional[Path]:
         """
+        Figure out the full path of the contract from the given ``key_contract_path``.
 
-        return self.path / "scripts"
+        For example, give it ``HelloWorld`` and it returns
+        ``<absolute-project-path>/contracts/HelloWorld.sol``.
 
-    @property
-    def tests_folder(self) -> Path:
-        """
-        The path to the ``tests/`` directory of the project.
+        Another example is to give it ``contracts/HelloWorld.sol`` and it also
+        returns ``<absolute-project-path>/contracts/HelloWorld.sol``.
```

Comment:
Would it be more helpful to change the example here to be a different folder than contracts? This allows for something like `another_contract_folder/HelloWorld.sol` and it would return `<absolute-project-path>/another_contract_folder/HelloWorld.sol`?

Reply:
Yeah, maybe if it said `<contracts-folder>`, it would be more obvious that it is a variable to the project. Good catch!